From c18cbdf3128ea46498925adcff31e04dbf3864c7 Mon Sep 17 00:00:00 2001
From: Ratchanan Srirattanamet <peathot@hotmail.com>
Date: Tue, 30 Jan 2024 14:09:00 +0700
Subject: [PATCH 0867/1085] iommu/bcm2712: don't allow building as module

Since bcm2712-iommu{,-cache}.c doesn't have usual module descriptors
such as `MODULE_LICENSE`, configuring this as 'M' fails the build with
`ERROR: modpost: missing MODULE_LICENSE() in <...>/bcm2712-iommu.o`.
Since it seems like the code is not intended to be built as a module
anyway (it registers the driver with `builtin_platform_driver()`), don't
allow building this code as a module.

Signed-off-by: Ratchanan Srirattanamet <peathot@hotmail.com>
---
 drivers/iommu/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/iommu/Kconfig
+++ b/drivers/iommu/Kconfig
@@ -495,7 +495,7 @@ config SPRD_IOMMU
 	  Say Y here if you want to use the multimedia devices listed above.
 
 config BCM2712_IOMMU
-       tristate "BCM2712 IOMMU driver"
+       bool "BCM2712 IOMMU driver"
        depends on ARM64 && ARCH_BCM
        select IOMMU_API
        help
